hcn/hcnnetwork.go

```diff
@@ -75,6 +75,30 @@ type HostComputeNetwork struct {
 	SchemaVersion SchemaVersion   `json:",omitempty"`
 }
 
+// NetworkResourceType are the 3 different Network settings resources.
+type NetworkResourceType string
+
+var (
+	// NetworkResourceTypePolicy is for Network's policies. Ex: RemoteSubnet
+	NetworkResourceTypePolicy NetworkResourceType = "Policy"
+	// NetworkResourceTypeDNS is for Network's DNS settings.
+	NetworkResourceTypeDNS NetworkResourceType = "DNS"
+	// NetworkResourceTypeExtension is for Network's extension settings.
+	NetworkResourceTypeExtension NetworkResourceType = "Extension"
+)
+
+// ModifyNetworkSettingRequest is the structure used to send request to modify an network.
+// Used to update DNS/extension/policy on an network.
+type ModifyNetworkSettingRequest struct {
+	ResourceType NetworkResourceType `json:",omitempty"` // Policy, DNS, Extension
+	RequestType  RequestType         `json:",omitempty"` // Add, Remove, Update, Refresh
+	Settings     json.RawMessage     `json:",omitempty"`
```

Comment:
Why not just make this `interface{}` and then do a type assertion in `ModifyNetworkSettings` that `Settings` is of type `PolicyNetworkRequest` today and it can grow as you add more types. But this way you only need the JSON serialization code in one spot.

Reply:
I don't anticipate this growing in the future (famous last words).  We should be using PolicyNetworkRequest for most changes.  I'm not sure why DNS and Extension aren't policies (maybe because of backwards compatibility).